src/resolvers/Mutation/updateUserTag.ts

```diff
@@ -0,0 +1,96 @@
+import { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { errors, requestContext } from "../../libraries";
+import { User, OrganizationTagUser } from "../../models";
+import {
+  USER_NOT_FOUND_ERROR,
+  USER_NOT_AUTHORIZED_ERROR,
+  TAG_NOT_FOUND,
+  NO_CHANGE_IN_TAG_NAME,
+  TAG_ALREADY_EXISTS,
+} from "../../constants";
+
+export const updateUserTag: MutationResolvers["updateUserTag"] = async (
+  _parent,
+  args,
+  context
+) => {
+  const currentUser = await User.findOne({
+    _id: context.userId,
+  }).lean();
+
+  // Checks whether currentUser exists.
+  if (!currentUser) {
+    throw new errors.NotFoundError(
+      requestContext.translate(USER_NOT_FOUND_ERROR.MESSAGE),
+      USER_NOT_FOUND_ERROR.CODE,
+      USER_NOT_FOUND_ERROR.PARAM
+    );
+  }
+
+  // Get the tag object
+  const existingTag = await OrganizationTagUser.findOne({
+    _id: args.input._id,
+  }).lean();
+
+  if (!existingTag) {
+    throw new errors.NotFoundError(
+      requestContext.translate(TAG_NOT_FOUND.MESSAGE),
+      TAG_NOT_FOUND.CODE,
+      TAG_NOT_FOUND.PARAM
+    );
+  }
+
+  // Boolean to determine whether user is an admin of organization of the tag folder.
+  const currentUserIsOrganizationAdmin = currentUser.adminFor.some(
+    (organization) =>
+      organization.toString() === existingTag!.organizationId.toString()
+  );
+
+  // Checks whether currentUser can update the tag
+  if (
+    !(currentUser.userType === "SUPERADMIN") &&
+    !currentUserIsOrganizationAdmin
+  ) {
+    throw new errors.UnauthorizedError(
+      requestContext.translate(USER_NOT_AUTHORIZED_ERROR.MESSAGE),
+      USER_NOT_AUTHORIZED_ERROR.CODE,
+      USER_NOT_AUTHORIZED_ERROR.PARAM
+    );
+  }
+
+  // Throw error if the new tag name is the same as the old one
+  if (existingTag!.name === args.input.name) {
+    throw new errors.ConflictError(
+      requestContext.translate(NO_CHANGE_IN_TAG_NAME.MESSAGE),
+      NO_CHANGE_IN_TAG_NAME.CODE,
+      NO_CHANGE_IN_TAG_NAME.PARAM
+    );
+  }
+
+  // Check if another tag with the new name exists under the same parent tag
+  const anotherTagExists = await OrganizationTagUser.exists({
```

Comment:
This check is wrong. What if `parentTagId` is null? You'd be checking against `tag.name` of tags which can belong to many other organizations. You have to include `organizationId` here, in this check.

Always think of all the possible edge cases, handle them gracefully, and only after that you should do what the function is supposed to do.

Reply:
Ahhh nice catch! Missed this one completely. Will try to be more aware about the same in the future.